node/remote/node.go

```diff
@@ -269,9 +271,14 @@ func (cp *Node) Tx(hash string) (*types.Tx, error) {
 }
 
 // Txs implements node.Node
+
 func (cp *Node) Txs(block *tmctypes.ResultBlock) ([]*types.Tx, error) {
-	txResponses := make([]*types.Tx, len(block.Block.Txs))
-	for i, tmTx := range block.Block.Txs {
+	txs := block.Block.Txs
+	if cp.ignoreConnectVoteExtensionTx && len(block.Block.Txs) > 0 {
+		txs = block.Block.Txs[1:]
```

Comment:
One issue that I have with this implementation is: what if we don't have vote extensions in all blocks? 

I think it would be better to instead skip transactions when trying to de-serialize them rather than here. So, inside the `Tx` method rather than `Txs`: if `ignoreConnectVoteExtensionTx` is true and we get a `wire type` error, then we ignore that transaction. Otherwise we proceed as usual. 

What do you think @hallazzang @rustcandy?

Reply:
It looks like always at the first place of txs according to the code @hallazzang found above. Do you think there's still possibility it's missing somewhere?

Reply:
Yes, for example in a chain that supports VoteExtensions but is not able to fetch the prices. I am not sure whether the transaction in that case would be there or not. For example a single-validator chain that only enables `connect` at height `20`: for all previous blocks there shouldn't be that kind of transaction. Or am I wrong?

Reply:
There's error on My localnet: is this the case? 
`3:48PM ERR oracle returned nil prices for vote extension; returning empty vote extension height=2601 module=server`
This case, there's still tx[0] exist.

Reply:
I think by simply checking the `wire` type error we also prevent any possible other case (e.g. multiple injected transactions)

Reply:
@RiccardoM Actually that was a valid concern. There is a configuration called `abci.vote_extensions_enable_height` inside `ConsensusParams` so I think this is true:

> For example a single-validator chain that only enables connect at height 20: for all previous blocks there shouldn't be that kind of transaction